src/scripts/main.js

```diff
@@ -249,6 +256,32 @@ import * as detailingFuncs from './modules/details/legacy';
         return Object.values(mice);
     }
 
+    // Extract map loot from a map
+    function getMapLoot(data, uncaught_only) {
+        const hint = {};
+        $.each(data.treasure_map.goals.item, (key, item) => {
+            hint[item.unique_id] = item.name;
+        });
+
+        if (uncaught_only) {
+            $.each(data.treasure_map.hunters, (key, hunter) => {
+                $.each(hunter.completed_goal_ids.item, (key, item_id) => {
+                    delete hint[item_id];
+                });
+            });
+        }
+
+        return Object.values(hint);
+    }
+
+
+    function getTreasureClue(data, uncaught_only, input_name) {
+        if(input_name == 'mice')
+            return getMapMice(data, uncaught_only);
+        if(input_name == 'items')
+            return getMapLoot(data, uncaught_only);
+    }
+
```

Comment:
Since this function is a duplicate of the other, we should simplify both by accessing the property based on what type of map it is.

```javascript
    /**
     * Extract goals from map (can be scav or treasure)
     * @param {Object} data Data response from managers/ajax/users/treasuremap.php
     * @param {boolean} remaining_only Boolean, if true, to filter out completed goals
     * @returns {string[]} Goal names. If on treasure map, it will be mice names and if
     * on scavenger it will be item names.
     */
    function getMapGoals(data, remaining_only) {
        const goalCategory = data.treasure_map.is_scavenger_hunt ? 'item' : 'mouse';
        const goals = {};
        $.each(data.treasure_map.goals[goalCategory], (key, goal) => {
            goals[goal.unique_id] = goal.name;
        });

        if (remaining_only) {
            $.each(data.treasure_map.hunters, (key, hunter) => {
                $.each(hunter.completed_goal_ids[goalCategory], (key, goal_id) => {
                    delete goals[goal_id];
                });
            });
        }

        return Object.values(goals);
    }
```